### DISCUSSION TITLE
pip install --editable .  didn't run in VS code

### DISCUSSION THREAD

**Discussion by foma4tune** [speechbrain/speechbrain#2603] [Q&A]
### Describe the bug

![pip install editable error in VS code](https://github.com/speechbrain/speechbrain/assets/62245121/f299e12d-0c09-49ff-b4cd-729230c29b3b)

Getting requirement to build editable did not run successfully.

Also the VS code did recognize cmdlet when i tried to "import speechbrain as sb"

### Expected behaviour

I expect it to run editable successful and to recognise the import of speechbrain as sb

**Comment by asumagic** [comment]
This error isn't sufficient to know what's going on, copy-pasting the entire result of the command would probably help.

**Comment by foma4tune** [comment]
Below are the issues. it will be appreciated if i get guide on resolving it and executing the speechbrain task with the various dataset and models

![speechbrain requirement installation issue](https://github.com/speechbrain/speechbrain/assets/62245121/6d37cded-7f7c-41b6-a9c2-4be53551fc19)
![speechrain editable issue](https://github.com/speechbrain/speechbrain/assets/62245121/cfc4df55-37d6-403a-a11a-6ff792a86295)
![command error](https://github.com/speechbrain/speechbrain/assets/62245121/51bee5a8-2be1-4f4d-a48b-789413aa9979)

**Comment by asumagic** [comment]
The error comes from a dependency which fails to build in the case. I see that one line mentions CPython 3.13, what Python version do you have installed?

If you are using a version newer than `3.12` it may make sense to use `3.11` or `3.12` in a virtual environment, because it would probably allow using pre-built wheels rather than having them built manually (which is what happens here, and fails).

If everything fails it might make more sense for you to use WSL and install SB under Linux, as Windows might be less tested and more difficult to support in general.
